ziti/ziti.go

```diff
@@ -29,177 +28,155 @@ import (
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
+	"github.com/go-openapi/runtime"
 	"github.com/michaelquigley/pfxlog"
 	"github.com/openziti/channel/v2"
 	"github.com/openziti/channel/v2/latency"
+	"github.com/openziti/edge-api/rest_client_api_client/current_api_session"
+	"github.com/openziti/edge-api/rest_client_api_client/service"
+	"github.com/openziti/edge-api/rest_model"
+	"github.com/openziti/edge-api/rest_util"
 	"github.com/openziti/foundation/v2/versions"
 	"github.com/openziti/identity"
 	"github.com/openziti/metrics"
 	"github.com/openziti/sdk-golang/ziti/edge"
-	"github.com/openziti/sdk-golang/ziti/edge/api"
-	"github.com/openziti/sdk-golang/ziti/edge/impl"
-	"github.com/openziti/sdk-golang/ziti/edge/posture"
+	"github.com/openziti/sdk-golang/ziti/edge/network"
 	"github.com/openziti/sdk-golang/ziti/sdkinfo"
 	"github.com/openziti/sdk-golang/ziti/signing"
 	"github.com/openziti/transport/v2"
 	cmap "github.com/orcaman/concurrent-map/v2"
 	"github.com/pkg/errors"
 	metrics2 "github.com/rcrowley/go-metrics"
 	"github.com/sirupsen/logrus"
+	"net/http"
 )
 
+type SessionType rest_model.DialBind
+
 const (
 	LatencyCheckInterval = 30 * time.Second
 	LatencyCheckTimeout  = 10 * time.Second
-)
 
-func SetApplication(theAppId, theAppVersion string) {
-	sdkinfo.SetApplication(theAppId, theAppVersion)
-}
+	ClientConfigV1 = "ziti-tunneler-client.v1"
+	InterceptV1    = "intercept.v1"
+
+	SessionDial = rest_model.DialBindDial
+	SessionBind = rest_model.DialBindBind
+)
 
+// Context is the main interface for SDK instances that may be used to authenticate, connect to services, or host
+// services.
 type Context interface {
+	// Authenticate attempts to use credentials configured on the Context to perform authentication. The authentication
+	// implementation used is configured via the Authenticator field on an Option struct provided during Context
+	// creation.
 	Authenticate() error
-	GetCurrentIdentity() (*edge.CurrentIdentity, error)
+
+	// SetAuthenticator sets the current Authenticator to use for subsequent authentication requests
+	SetAuthenticator(authenticator rest_util.Authenticator)
```

Comment:
these changes make me think the version should change. i can't quite tell if we're bumping this rev or not?

Reply:
Yes, we should!